### DISCUSSION TITLE
关于pdf显示的问题

### DISCUSSION THREAD

**Discussion by yorelll** [obster-y/XJTU-thesis#92] [Q&A] [answered]
请问如何使用内置的pdf的显示，我在选项配置里设置了内置pdf显示，但在编译后还是只能通过外置的pdf显示，我通过一个全英文的tex文件测试是可以内置显示的，请问这种问题如何解决呢？

**Comment by obster-y** [comment]
给的信息不够清楚，以及这是软件使用的问题，基本上和模板没关系

如果想问，什么平台什么编辑器什么设置最好能说一下

如果是用 latexmk，有参数可以改

**Reply by yorelll** [reply]
感谢您的回复，我使用的是windows平台下的texstudio编辑器，编译器选择是latexmk，默认的pdf查看器是内置pdf查看器，请问如何修改可以使其内置编译显示呢？

**Reply by yorelll** [reply]
texstudio版本是2022

**Reply by obster-y** [reply]
这我也不是很清楚，很久没用 texstudio 了，建议去别的地方问问，或者看看 texstudio 的帮助。也许和 latexmkrc 里的 $preview_mode 有关。可以查看 https://www.mankier.com/1/latexmk

**Reply by yorelll** [reply]
好的好的，感谢！

**Comment by sikouhjw** [comment]
应该是
https://github.com/obster-y/XJTU-thesis/blob/d80e4bb7d9f2b7f0b94d528bd38e46f01e1da44a/latexmkrc#L6
的行为，`$preview_mode = 1` 等同于 `-pv`，`-pv` 的解释是
> Run file previewer. If the -view option is used, this will select the kind of file to be previewed
> (.dvi, .ps or .pdf). Otherwise the viewer views the "highest" kind of file selected, by the -dvi, -ps,
> -pdf, -pdfps options, in the order .dvi, .ps, .pdf (low to high). If no file type has been selected, the
> dvi previewer will be used. This option is incompatible with the -p and -pvc options, so it turns
> them off.

说明 latexmk『帮你』用 pdf 阅读器打开了 pdf，也就是『外置』pdf 阅读器。

想要改变这个行为，将 `$preview_mode = 0` 就可以了。

**Reply by yorelll** [reply]
是的，我阅读了您推荐的手册，当preview_mode取非零的时候，会调用一个预览器来显示，但是我将其设置为0，则不会调用任何预览器（包括内置），所以目前稍优的策略是使用sumatra此种可以不关闭文件即可写入的外置pdf预览软件，在分屏操作下可以实现编译的实时预览，但缺点无法通过源码跳转pdf，相反也无法通过pdf跳转源码。我可能还需要研究一下是否有其他更好的方案，再次感谢！

**Reply by sikouhjw** [reply]
> 但是我将其设置为0，则不会调用任何预览器（包括内置）

texstudio 有个按钮是『构建并查看』，这个按钮应该包含了 `latexmk` 和使用『内置』pdf 阅读器两种功能。

> 但缺点无法通过源码跳转pdf，相反也无法通过pdf跳转源码。

可以从《install-latex-guide-zh-cn》[^1]找到如何设置 texstudio 与 SumatraPDF 之间的正反跳转的方法。

[^1]: https://github.com/OsbertWang/install-latex-guide-zh-cn

**Reply by yorelll** [reply]
好的好的，我来研究一下，谢谢。

**Comment by yorelll** [comment, marked as answer]
一个解决方案是使用xelatex编译，这样可以很好的解决无法使用内部pdf预览器的问题。再次感谢作者的回复！
